### PR TITLE
[Documentation] Code review guidelines & learning resources

### PR DESCRIPTION
**Pull request by Olshansk**
- Add GPokT to the main README
- Add a foundation "code guidelines" only containing examples of comments
- Add details on tending resolving comments and reviewing PRs
- Add resources related to trees

### CODE REVIEW

**Comment by Olshansk** [comment]
@red-0ne @0xRampey @dylanlott @h5law Friendly bump on this.

File: docs/development/CODE_GUIDELINES.md [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+func (persistencePSP *persistencePeerstoreProvider) GetUnstakedPeerstore() (typesP2P.Peerstore, error) {
+  // ...
+}
+```
+
+## Exposing functions for testing purposes
+
+If possible, move the function into a separate file that has `//go:build test` at the top like so:
+
+![go:build test](https://github.com/pokt-network/pocket/assets/1892194/e7f921c7-6830-4aa6-afca-aef9c6cabbc6)https://github.com/pokt-network/pocket/assets/1892194/e7f921c7-6830-4aa6-afca-aef9c6cabbc6
```
Comment (dylanlott) [inline docs/development/CODE_GUIDELINES.md:36]:
```suggestion
![go:build test](https://github.com/pokt-network/pocket/assets/1892194/e7f921c7-6830-4aa6-afca-aef9c6cabbc6)https://github.com/pokt-network/pocket/assets/1892194/e7f921c7-6830-4aa6-afca-aef9c6cabbc6

For further reading, please see [Testutils](https://www.notion.so/pocketnetwork/Testutils-9cba9010e18447248e9daa8a3b87e3f2)
```

File: docs/learning/README.md [hunk 2]
```diff
@@ -11,9 +11,10 @@ _This is a live document on how to get ramped up on all the knowledge you need t
 - [📚 Technical References](#-technical-references)
   - [Pocket Specific](#pocket-specific)
   - [Consensus](#consensus)
-  - [Merkle Trees](#merkle-trees)
+  - [🌲 Trees](#-trees)
```
Comment (dylanlott) [inline docs/learning/README.md:14]: 🌳 💯 🌴

File: docs/learning/README.md [hunk 3]
```diff
@@ -220,12 +254,12 @@ This is a general set of technical links and recommended reading our team has fo
...
-- [Alin Tomescu](https://alinush.github.io/)
-- [Joachim Neu](https://www.jneu.net/)
-  - See the articles under the **Technical reports** section
-- [Decentralized Thoughts by Ittai Ibrahim](https://decentralizedthoughts.github.io/about-ittai); the RSS feed is available [here](https://decentralizedthoughts.github.io/feed.xml)
+- [Mike's blog](https://morourke.org/); yours truly
+- [Vitalik's blog](https://vitalik.ca/categories/blockchains.html); hopefully Vitalik needs no introduction if you're reading this
+- [Alin Tomescu](https://alinush.github.io/); Alin the is one of the founding engineers at Aptos Labs
+- [Joachim Neu](https://www.jneu.net/); focus on the articles under the **Technical reports** section
+- [Decentralized Thoughts by Ittai Ibrahim](about-ittai); The best blog about everything consensus blockchain related by one of the leading researchers in the field (the RSS feed is available [here](https://decentralizedthoughts.github.io/feed.xml))
+- [Polymer Labs](https://polymerlabs.medium.com/); polymer has very easy to digest and easy to read blogs on IBC and interoperability
```
Comment (dylanlott) [inline docs/learning/README.md:262]:
```suggestion
- [Polymer Labs](https://polymerlabs.medium.com/); polymer has very easy to digest and easy to read blogs on IBC and interoperability
- [Julia Evans' blog](https://jvns.ca/)
```

**Review by dylanlott — APPROVED** [review]
LGTM  ✅

**Pull request merged**